src/Microsoft.Framework.PackageManager/Restore/NuGet/NuGetv2Feed.cs

```diff
@@ -232,17 +240,49 @@ public PackageInfo BuildModel(string id, XElement element)
             });
         }
 
+        private void EnsureValidFindPackagesResponse(XDocument doc)
+        {
+            if (!string.Equals(doc.Root.GetDefaultNamespace().ToString(), _defaultNamespace,
+                StringComparison.OrdinalIgnoreCase))
+            {
+                throw new FormatException("The XML is not a valid NuGet v2 service response to FindPackagesById() request");
```

Comment:
Yeah, you can say it is an invalid operation if you send HTTP requests to a server that responds with wrong contents. However, I guess using `FormatException` can give users more intuitive understanding of what is wrong here.

What's the principle you are following when you decide to use `InvalidOperationException`?